vmdb/app/controllers/api_controller/providers.rb

```diff
@@ -0,0 +1,158 @@
+class ApiController
+  module Providers
+    TYPE_ATTR         = "type"
+    CREDENTIALS_ATTR  = "credentials"
+    AUTH_TYPE_ATTR    = "auth_type"
+    DEFAULT_AUTH_TYPE = "default"
+    RESTRICTED_ATTRS  = [TYPE_ATTR, CREDENTIALS_ATTR] + %w(zone zone_id)
+
+    def create_resource_providers(type, _id, data = {})
+      if data.key?("id") || data.key?("href")
+        raise BadRequestError,
+              "Resource id or href should not be specified for creating a new #{type}"
```

Comment:
do you want to raise an error, or just ignore?

``ruby
def create_resource_providers(type, _id, data = {})
  create_provider(data.except(%w(id href)))
end
```